### PR TITLE
ISS-107 # file upload zerocode demo tests

### PR DESCRIPTION
**Pull request by nirmalchandra**

### CODE REVIEW

File: src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+        return serverResponse;
+    }
+
+    private RequestBuilder createFileUploadRequestBuilder(String httpUrl, String methodName, String reqBodyAsString) throws IOException {
+        String fileName = getFileNameFromPayLoad(reqBodyAsString);
+        FileBody file = new FileBody(new File(getAbsPath(fileName)));
+
+        HttpEntity reqEntity = MultipartEntityBuilder.create()
+                .addPart("file", file)
```
Comment (authorjapps) [inline src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java]: "`file`" is hardcoded. Instead can you pick it from the request?
Though server might expect another `key`, although most of them use `file`.
You can do it like below-
```
{
   "name": "file",
   "fileName": "folder/abcd.pdf"
}
or
{
   "file": "file:folder/abcd.pdf"  // "file": "content:folder/abcd.pdf"
}
```

File: src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                .setBoundary("" + System.currentTimeMillis() + LocalDateTime.now())
```
Comment (authorjapps) [inline src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java]: `setBoundary(...)` argument is good to **default** it to `current time`, but in this example allow it to pick from the request, if not present, then default it to current time(which is done already)

File: src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+        } else {
+
+            httpResponse = httpclient.execute(requestBuilder.build());
+        }
+
+        HttpEntity entity = httpResponse.getEntity();
+        Response serverResponse = Response
+                .status(httpResponse.getStatusLine().getStatusCode())
+                .entity(entity != null ? IOUtils.toString(entity.getContent()) : null)
```
Comment (authorjapps) [inline src/test/java/com/springboot/testrunner/httpclient/ProjectApacheHttpClient.java]: 👍

File: src/test/resources/integration_tests/upload_file/file_upload_test.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            "name": "upload_a_pdf_file",
+            "url": "/api/abc-bank/upload",
+            "operation": "POST",
+            "request": {
+                "headers": {
+                    "content-type": "multipart/form-data"
+                },
+                "body": {
+                    "file": "project_files/my_zip_file.zip"  // Allows 'test/resources' path or absolute path
```
Comment (authorjapps) [inline src/test/resources/integration_tests/upload_file/file_upload_test.json]: Can you handle multiple files ?

**Comment by nirmalchandra** [comment]
All review comments has been fixed. Please approve it now !

**Review by authorjapps — APPROVED** [review]

**Pull request merged**